core/chains/evm/client/rpc_client.go

```diff
@@ -195,14 +241,16 @@ func (r *rpcClient) Dial(callerCtx context.Context) error {
 	}
 	lggr.Debugw("RPC dial: evmclient.Client#dial")
 
-	wsrpc, err := rpc.DialWebsocket(ctx, r.ws.uri.String(), "")
-	if err != nil {
-		promEVMPoolRPCNodeDialsFailed.WithLabelValues(r.chainID.String(), r.name).Inc()
-		return r.wrapRPCClientError(pkgerrors.Wrapf(err, "error while dialing websocket: %v", r.ws.uri.Redacted()))
-	}
+	if r.ws.uri != (url.URL{}) {
```

Comment:
why this check?

Reply:
This check is needed to get tests passing where secondary URLs are used, since setting up send only nodes with http in this case also provides this empty url (we don't pass anything specifying if it's send only). Without the check, the ws will be dialled and break the test.